yt_dlp/extractor/ilpost.py

```diff
@@ -35,34 +37,75 @@ def _real_extract(self, url):
+            r'{"props":{"pageProps":', webpage, 'metadata', display_id)
+        episode_id = endpoint_metadata['data']['data']['episode']['data'][0]['id']
+        podcast_id = traverse_obj(endpoint_metadata, ('data', 'data', 'episode', 'data', 0, 'parent', 'id'))
+        podcast_metadata = traverse_obj(endpoint_metadata, ('data', 'data', 'episode', 'data', 0))
 
-        episode = traverse_obj(podcast_metadata, (
-            'data', 'postcastList', lambda _, v: str(v['id']) == episode_id, {dict}), get_all=False)
+        episode = podcast_metadata
         if not episode:
             raise ExtractorError('Episode could not be extracted')
 
         return {
-            'id': episode_id,
-            'display_id': display_id,
-            'series_id': podcast_id,
-            'vcodec': 'none',
+            'id': str(episode_id),
+            'display_id': str(display_id),
+            'series_id': str(podcast_id),
             **traverse_obj(episode, {
                 'title': ('title', {str}),
-                'description': ('description', {str}),
-                'url': ('podcast_raw_url', {url_or_none}),
+                'url': ('episode_raw_url', {url_or_none}),
                 'thumbnail': ('image', {url_or_none}),
+                'description': ('content_html', {str}),
                 'timestamp': ('timestamp', {int_or_none}),
                 'duration': ('milliseconds', {float_or_none(scale=1000)}),
-                'availability': ('free', {lambda v: 'public' if v else 'subscriber_only'}),
+                'availability': ('access_level', {lambda v: 'public' if v else 'subscriber_only'}),
             }),
         }
+
+
+class IlPostPodcastIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?ilpost\.it/podcasts/(?P<id>[a-zA-Z0-9\-]+)/?$'
+    _TESTS = [{
+        'url': 'https://www.ilpost.it/podcasts/morning/',
+        'info_dict': {
+            'id': 'morning',
+            'display_id': 'morning',
+            'title': 'Morning',
+            'series': 'Morning',
+            'season_number': 1,
+        },
+        'playlist_mincount': 20,
+    }, {
+        'url': 'https://www.ilpost.it/podcasts/basaglia-e-i-suoi/',
+        'info_dict': {
+            'id': 'basaglia-e-i-suoi',
+            'display_id': 'basaglia-e-i-suoi',
+            'title': 'Basaglia e i suoi',
+            'series': 'Basaglia e i suoi',
+            'season_number': 1,
+        },
+        'playlist_mincount': 5,
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_valid_url(url).group('id')
+        data = self._download_json(PODCAST_API % display_id, display_id)
+
+        entries = [{
+            '_type': 'url',
+            'url': episode['url'],
+            'title': episode.get('title'),
+            'description': episode.get('content_html'),
+            'series': traverse_obj(data, ('data', 0, 'parent', 'title')),
+            'season_number': 1,
+            'episode_number': episode['id'],
```

Comment:
is this an actual episode number int, e.g. `1` `5` `10` etc? or is some other numeric id?